### PR TITLE
Restore delete queue to handle the product deletion

### PR DESCRIPTION
**Pull request by phonglynosto**
Restore delete queue to handle the product deletion

## Description
This commit will revert the commit 99293cfe82aedae644b3377d2bd19cc1bf9a198a and make some modification to handle the product deletion

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] I have assigned the correct milestone or created one if non-existent.
- [ ] I have correctly labeled this pull request.
- [ ] I have linked the corresponding issue in this description.
- [ ] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [ ] I have checked the base branch of this pull request
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Api/Data/ProductIndexerIgnoranceInterface.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2023 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+namespace Nosto\Tagging\Api\Data;
+
+
+interface ProductIndexerIgnoranceInterface
```
Comment (supercid) [inline Api/Data/ProductIndexerIgnoranceInterface.php:39]: Perhaps ignorance is not the best name for this feature. How about changing to just ignore?

File: Model/Indexer/ProductIndexer.php [hunk 2]
```diff
@@ -128,6 +166,58 @@ public function doIndex(Store $store, array $ids = [])
...
+                $indexerExcludeRepository = $this->excludeRepositoryFactory->create();
+
+                $items = $indexerExcludeRepository->search($searchCriteria)->getItems();
+
+                $ignoreIds = array_map(function ($item) {
+                    return $item->getEntityId();
+                }, $items);
+
+                $givenIds = array_diff($givenIds, $ignoreIds);
+                $indexerExcludeRepository->deleteByIds($ignoreIds);
```
Comment (supercid) [inline Model/Indexer/ProductIndexer.php:198]: Can be extracted to a separate method. If I understood this correctly, you're removing the given ID's from the exclude table. In which situation would I add twice to that exclude table?

**Review by supercid — CHANGES_REQUESTED** [review]

File: Model/Indexer/ProductIndexer.php (hunk 2 above)
Comment (phonglynosto) [reply Model/Indexer/ProductIndexer.php:198]: Right now I don think any situation you can add twice into table but at least I don want table bigger

File: Model/Service/Sync/Delete/AsyncBulkPublisher.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        return self::BULK_SIZE;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBulkDescription(): string
+    {
+        return sprintf('Delete %d Nosto products', 2);
```
[the remark this answers is not in the document]
Comment (phonglynosto) [reply Model/Service/Sync/Delete/AsyncBulkPublisher.php:67]: That is from old commit, sorry does not check it carefully

**Review by supercid — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 All notable changes to this project will be documented in this file. This project adheres to Semantic Versioning.
 
+### 7.2.5
+* Restore `nosto_product_sync.delete` message queue consumer to handle product deletion
```
[the remark this answers is not in the document]
Comment (supercid) [reply CHANGELOG.md:4]: https://github.com/Nosto/wiki-docs/pull/101

**Pull request merged**